js/ui/marker.js

```diff
@@ -13,6 +13,10 @@ var Point = require('point-geometry');
  * @param {HTMLElement=} element DOM element to use as a marker (creates a div element by default)
  * @param {Object=} options
  * @param {PointLike=} options.offset The offset in pixels as a [`PointLike`](#PointLike) object to apply relative to the element's top left corner. Negatives indicate left and up.
+ * @param {boolean=} [options.pitchScale=false] Whether the marker should scale smaller into the background or larger into the foreground when the map is pitched and rotated
+ * @param {boolean=} [options.zoomScale=false] Whether the marker should scale with the map as you zoom in and out
+ * @param {number=} [options.nativeZoom=map.getZoom()] The native zoom level of the marker. If the map is at this zoom the marker will not be scaled. If zoom is greater the marker will be enlarged. If zoom is less the marker will be shrunk.
+ * @param {number=} [options.zoomFactor=1.6] The multiple by which to scale the size of the marker between each zoom level
```

Comment:
The problem, though, is that in in order to do that you have to calculate the desired value of the function at stops 1 and 20 anyway which defeats the purpose of delegating the calculations to the function in the first place.

Eg to get the marker to scale with the map 1:1 in the zoom range [1, 20] you'd have to use

```
stops: [
    [1, Math.pow(base, 1-map.getZoom())]
    [map.getZoom(), 1],
    [20, Math.pow(base, 20-map.getZoom())]
]
```

Whereas without using `mapbox-gl-function` you just need to set the scale to

```
Math.pow(base, currentZoom - nativeZoom)
```